### PR TITLE
feat: Map and Label Remove Tests

### PR DESCRIPTION
**Pull request by RODO94**
## What does this PR do?

Following previous work on Map and Label testing, I am continuing through and testing the Remove feature.

To test the feature being deleted the map, I am accessing the `drawgeojson` attribute of `<my-map>` seemed a good way to confirm it has been deleted from the map itself. Open to alternative suggestions though.

```ts
    const mapTwo = getByTestId("map-and-label-map");

    expect(mapTwo).toHaveAttribute(
      "drawgeojsondata",
      `{"type":"FeatureCollection","features":[]}`
    );
```

### CODE REVIEW

File: editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.test.tsx [hunk 1]
```diff
@@ -347,10 +347,62 @@ describe("copy feature select", () => {
...
+    const tabOnePanel = getByRole("tabpanel", { name: /Tree 1/ });
+
+    const removeButton = getByRole("button", { name: "Remove" });
+
+    await user.click(removeButton);
+
+    expect(tabOne).not.toBeInTheDocument();
+    expect(tabOnePanel).not.toBeInTheDocument();
+  });
+  it("removes a feature from the form - multiple features", async () => {
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.test.tsx:366]:
```suggestion
  });
  
  it("removes a feature from the form - multiple features", async () => {
```

Comment (DafyddLlyr) [reply editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.test.tsx:366]: nit: Spacing - we should probably find a lint rule for this tbh!

File: editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.test.tsx [hunk 2]
```diff
@@ -347,10 +347,62 @@ describe("copy feature select", () => {
...
+    const removeButton = getByRole("button", { name: "Remove" });
+
+    await user.click(removeButton);
+
+    const mapTwo = getByTestId("map-and-label-map");
+
+    expect(mapTwo).toHaveAttribute(
+      "drawgeojsondata",
+      `{"type":"FeatureCollection","features":[]}`
+    );
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.test.tsx:404]: Great solution 👌 

How do we know that `{"type":"FeatureCollection","features":[]}` wasn't the value all along and that it's actually changed?

**Review by DafyddLlyr — CHANGES_REQUESTED** [review]
One thing to take a look at in the `"removes a feature from the map"` test 👍

File: editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.test.tsx [hunk 3]
```diff
@@ -347,10 +347,62 @@ describe("copy feature select", () => {
...
+    const { getByTestId, getByRole, user } = setup(<MapAndLabel {...props} />);
+    const map = getByTestId("map-and-label-map");
+
+    addFeaturesToMap(map, [point1]);
+
+    const removeButton = getByRole("button", { name: "Remove" });
+
+    await user.click(removeButton);
+
+    const mapTwo = getByTestId("map-and-label-map");
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.test.tsx:399]: nit: I think that `mapTwo` is a little unclear - we should just use `let` to define the map originally.

**Review by DafyddLlyr — CHANGES_REQUESTED** [review]

File: editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.test.tsx (hunk 2 above)
Comment (RODO94) [reply editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.test.tsx:404]: Ah sick, I'll put in a population check in an earlier test confirming when you had add a feature, it gets added to this

File: editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.test.tsx [hunk 4]
```diff
@@ -205,13 +210,16 @@ test.todo("an error displays if the maximum number of items is exceeded");
...
 
+    let map = getByTestId("map-and-label-map");
     addFeaturesToMap(map, [point1]);
 
     const firstTabPanel = getByTestId("vertical-tabpanel-0");
 
     expect(firstTabPanel).toBeVisible();
+
+    map = getByTestId("map-and-label-map");
+    expect(map).toHaveAttribute("drawgeojsondata", mockFeaturePointObj);
```
Comment (RODO94) [inline editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.test.tsx:222]: @DafyddLlyr happy path `drawgeojsondata` check added here

**Review by DafyddLlyr — APPROVED** [review]

**Pull request merged**